### PR TITLE
[Azure.Identity] Add e2e testing instructions for Arc MI

### LINKED ISSUE #14371: Create Identity e2e tests for running in Azure Arc
Tracking item to create new /sdk/e2e Azure.Identity smoke tests for running in Azure Arc hosted environment.

1. Add new /sdk/e2e tests for Azure Arc
2. Update Parent issue https://github.com/Azure/azure-sdk/issues/1908 with instructions for Test team

### PR DESCRIPTION
**Pull request by mccoyp**
Closes #14371.

### CODE REVIEW

File: sdk/identity/azure-identity/tests/managed-identity-live/azure_arc.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+Set the environment variable `AZURE_IDENTITY_TEST_VAULT_URL` to the vault URI of your key vault.
+
+Install `requirements.txt`:
+```
+pip install -r requirements.txt
+```
+Run the managed identity tests:
+```
+pytest test_managed_identity_live.py
+pytest test_managed_identity_live_async.py
```
Comment (chlowell) [inline sdk/identity/azure-identity/tests/managed-identity-live/azure_arc.md:49]: You can run both with a single command:
```suggestion
pytest -k test_managed_identity_live
```
And this should be done twice, once each for Python 2.7 and 3.x.

File: sdk/identity/azure-identity/tests/managed-identity-live/azure_arc.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
+# Testing azure-identity in Azure Arc
+
+## Prerequisite tools
+
+1. A non-Azure Windows or Linux VM.
+2. Administrator privileges on the VM.
+3. An Azure Key Vault.
+4. Python 3.5+
```
Comment (chlowell) [inline sdk/identity/azure-identity/tests/managed-identity-live/azure_arc.md:8]: We also need to test 2.7.

**Comment by catalinaperalta** [comment]
Once this is approved and merged, can you please also copy and paste these instructions in this issue (https://github.com/Azure/azure-sdk/issues/1908) for the vendor team?

File: sdk/identity/azure-identity/tests/managed-identity-live/azure_arc.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+2. Administrator privileges on the VM.
+3. An Azure Key Vault.
+4. Python 2.7 and 3.5+
+
+### Install Azure Arc on the VM
+
+> **Note:** You must be in your VM to install Azure Arc.
+
+1. Create an Azure Arc server resource on the [Azure Portal](https://portal.azure.com).
+2. Choose to add an existing server using an interactive script.
```
Comment (catalinaperalta) [inline sdk/identity/azure-identity/tests/managed-identity-live/azure_arc.md:15]: Not sure if a user who isn't familiar with the Arc section of the portal will know where to find the add server section. Maybe add something saying that it comes after selecting "Add to your existing infrastructure"?

File: sdk/identity/azure-identity/tests/managed-identity-live/azure_arc.md [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+6. Installation should be complete. Restart your VM to complete your environment setup.
+7. After restarting, check your environment by searching for environment variables named `IDENTITY_ENDPOINT` and 
+`IMDS_ENDPOINT`. If they are not present, or don't resemble `http://localhost:40342/metadata/identity/oauth2/token` or 
+`http://localhost:40342` respectively, you may need to wait a short while or try restarting the VM again.
+8. If using a Linux VM, run the following commands (using your user name for `<user>`) to gain necessary privileges:
+```
+sudo usermod -a -G himds <user>
+sudo setfacl -m "g:himds:r-x" /var/opt/azcmagent/tokens/
+sudo setfacl -m "g::r-x" /var/opt/azcmagent/tokens/
+```
```
Comment (catalinaperalta) [inline sdk/identity/azure-identity/tests/managed-identity-live/azure_arc.md:29]: I had to restart after applying these changes, did you also have to do that? If so, I'd mention that a restart might be necessary after this.

Comment (mccoyp) [reply sdk/identity/azure-identity/tests/managed-identity-live/azure_arc.md:29]: That's a good point; I do remember having to restart. I'll add that 👍

File: sdk/identity/azure-identity/tests/managed-identity-live/azure_arc.md (hunk 3 above)
Comment (mccoyp) [reply sdk/identity/azure-identity/tests/managed-identity-live/azure_arc.md:15]: I actually haven't seen a prompt like that from creating new "Servers - Azure Arc" resources. I only see a choice between "Add servers using interactive script" and "Add servers at scale" -- I'll be more specific about how to create the resource to reflect this

**Review by catalinaperalta — APPROVED** [review]
LGTM

File: sdk/identity/azure-identity/tests/managed-identity-live/azure_arc.md [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+Set the environment variable `AZURE_IDENTITY_TEST_VAULT_URL` to the vault URI of your key vault.
+
+Install `requirements.txt`:
+```
+pip install -r requirements.txt
+```
+Run the managed identity tests, using the below command once with Python 2.7 and once with Python 3.5+:
+```
+pytest -k managed_identity_live
```
Comment (chlowell) [inline sdk/identity/azure-identity/tests/managed-identity-live/azure_arc.md:49]: Don't you need to add a Key Vault access policy for the identity first?

File: sdk/identity/azure-identity/tests/managed-identity-live/azure_arc.md [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+5. Once the script has been downloaded, run the script on your machine with administrator privileges.
+6. If using a Linux VM, run the following commands (using your user name for `<user>`) to gain necessary privileges:
+```
+sudo usermod -a -G himds <user>
+sudo setfacl -m "g:himds:r-x" /var/opt/azcmagent/tokens/
+sudo setfacl -m "g::r-x" /var/opt/azcmagent/tokens/
+```
+7. Arc setup should now be complete. Restart your VM to finalize your environment setup.
+8. After restarting, check your environment by searching for environment variables named `IDENTITY_ENDPOINT` and 
+`IMDS_ENDPOINT`. If they are not present, or don't resemble `http://localhost:40342/metadata/identity/oauth2/token` and
```
Comment (chlowell) [inline sdk/identity/azure-identity/tests/managed-identity-live/azure_arc.md:29]: Just curious--this suggests Arc may set unusable values for the variables. Have we seen that happen?

**Review by chlowell — APPROVED** [review]

File: sdk/identity/azure-identity/tests/managed-identity-live/azure_arc.md (hunk 6 above)
Comment (mccoyp) [reply sdk/identity/azure-identity/tests/managed-identity-live/azure_arc.md:29]: We haven't to my knowledge but I just wanted to make sure our bases are covered. I would imagine they would either show up correctly or not at all, assuming an Arc environment can't also be configured as another environment where `IDENTITY_ENDPOINT` would get set

**Pull request merged**